index.js

```diff
@@ -18,41 +22,36 @@ var suffix = new RegExp(qs)
 module.exports = Nanorouter
 
 function Nanorouter (opts) {
+  if (!(this instanceof Nanorouter)) return new Nanorouter(opts)
   opts = opts || {}
+  this.router = wayfarer(opts.default || '/404')
+}
+
+Nanorouter.prototype.on = function onRoute (routename, listener) {
+  assertRoutename(routename)
+  routename = routename.replace(/^[#/]/, '')
+  this.router.on(routename, listener)
+}
+
+Nanorouter.prototype.emit = function emitRoute (routename) {
+  assertRoutename(routename)
+  routename = pathname(routename, isLocalFile)
+  return this.router.emit(routename)
+}
+
+Nanorouter.prototype.match = function matchRoute (routename) {
+  assertRoutename(routename)
+  routename = pathname(routename, isLocalFile)
+  return this.router.match(routename)
+}
 
-  var router = wayfarer(opts.default || '/404')
-  var curry = opts.curry || false
-  var prevCallback = null
-  var prevRoute = null
-
-  emit.router = router
-  emit.on = on
-  return emit
-
-  function on (routename, listener) {
-    routename = routename.replace(/^[#/]/, '')
-    router.on(routename, listener)
-  }
-
-  function emit (route) {
-    if (!curry) {
-      return router(route)
-    } else {
-      route = pathname(route, isLocalFile)
-      if (route === prevRoute) {
-        return prevCallback()
-      } else {
-        prevRoute = route
-        prevCallback = router(route)
-        return prevCallback()
-      }
-    }
-  }
+function assertRoutename (routename) {
```

Comment:
I think we should remove this function, and write these statements inline. This has two benefits:
1. writing duplicate code compresses better. So while having more code in source, it'll probably end up being less code sent over the wire.
2. this no longer works with static transforms such as [unassertify](https://github.com/unassert-js/unassertify) and [tinyify](https://github.com/browserify/tinyify).

Hope this makes sense!

Reply:
Yes, definitely. I'll just use assert then; that's basically already in use every setup.